### PR TITLE
Fix Fountain of Youth yield as per wiki

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
- F of Y "Tile provides yield without assigned population"
- Made that unique work on terrain
- Byzantium startBias fix - a pity - "Coats" sounds so nice. Should have done a terrain "Coats" instead.

Open question: This works to provide yield (and prevent worker assignment) when within the workable radius of a city and owned. How it should behave when owned by - 4 tiles from city center - I'm not sure.

### CODE REVIEW

**Comment by xlenstra** [comment]
If I am not mistaken (but I very well could be), in general, happiness yields from all tiles inside your border is always provided, regardless of whether the tile is worked or not. For the fountain of youth, this turns out to be equivalent to have all tile yields always be provided, as it only produces happiness. In general, this is not the case, so there is an argument that this implementation isn't necessary when that behavior is implemented correctly

**Comment by SomeTroglodyte** [comment]
> happiness yields from all tiles inside your border is always provided

OK that would be a whole different kettle of fish, implementation-wise. And this actually provides more than 10 happiness: 10 for the json entry, +1 happy just because it's a Nat Won, and +1 P +1 F you get for the turnsInto Plains...

OK, I suspect this is not original C5 (the game, not the explosive) behaviour -> but it shows what _could_ be done with that otherwise unused unique. Was that one done for some mod I don't know?

**Comment by logicminimal** [comment]
what if there is also a yield when its being worked?

File: android/assets/jsons/Civ V - Vanilla/Nations.json [hunk 1]
```diff
@@ -804,7 +804,7 @@
 		"name": "Byzantium",
 		"leaderName": "Theodora",
 		"adjective": ["Byzantine"],
-		"startBias": ["Coats"],
```
Comment (yairm210) [inline android/assets/jsons/Civ V - Vanilla/Nations.json:807]: Maybe they just happen to really like coats. Fair enough.

**Comment by SomeTroglodyte** [comment]
OK - decision time. 11 Happiness displayed in two separate places +1 production +1 food, as long as in workable and owned space, but unable to be worked for any extra yield, is partially right but overall still wrong.

But the unique would be closer to what it advertises - "tile yield" does sound like terrain just as much as it sounds like an improvement - and it works as it is designed to do.

I vote (pushed version) the code to support the "yield without worker" unique on terrain stays in, but the Fountain doesn't get it. A potential fix to FoY would then be another PR and technically quite a bit different.

To reiterate what I now believe it should do:
- +1 Happiness when found and ever after (currently works)
- +10 Happiness _instead of_ +1 when in owned territory, no matter distance from owning city (nope)
- Cannot be assigned a worker (nope - currently you can, and then it totals +11 Happiness)

@xlenstra  or @ravignir - right or wrong?

File: android/assets/jsons/Civ V - Vanilla/Nations.json (hunk 1 above)
Comment (SomeTroglodyte) [reply android/assets/jsons/Civ V - Vanilla/Nations.json:807]: Calls for a mod where Byzantium always spawns near a Natural Wonder named "Coat Rack" and gets a unique Worker replacement named "Coats" and a Warrior replacement named "Coåts" and an Archer replacement named "Cøats" and a Scout replacement named "Çoats" - all having the exact same icon and pixel unit graphic... and so on...

**Comment by ravignir** [comment]
+10 Happiness when within borders (+20 as sPAIN), +1 when discovered (+2 when sPAIN). Natural wonders stats should override base terrain stats.

@SomeTroglodyte Off-topic: I was supposed to take a break from Unciv 😅

**Comment by xlenstra** [comment]
> OK - decision time. 11 Happiness displayed in two separate places +1 production +1 food, as long as in workable and owned space, but unable to be worked for any extra yield, is partially right but overall still wrong.
> 
> But the unique would be closer to what it advertises - "tile yield" does sound like terrain just as much as it sounds like an improvement - and it works as it is designed to do.
> 
> I vote (pushed version) the code to support the "yield without worker" unique on terrain stays in, but the Fountain doesn't get it. A potential fix to FoY would then be another PR and technically quite a bit different.

Agree

> To reiterate what I now believe it should do:
> 
>     * +1 Happiness when found and ever after (currently works)
> 
>     * +10 Happiness _instead of_ +1 when in owned territory, no matter distance from owning city (nope)
> 
>     * Cannot be assigned a worker (nope - currently you can, and then it totals +11 Happiness)
> 
> 
> xlenstra or ravignir - right or wrong?

Based on a quick test I did, the first two seem correct: the +10 overrides the normal +1. Didn't have the time to test the third

**Comment by SomeTroglodyte** [comment]
A break? Sorry to interrupt, then. Brakes are important! Especially downhill...

**Comment by xlenstra** [comment]
> Based on a quick test I did, the first two seem correct: the +10 overrides the normal +1. Didn't have the time to test the third

Also tested the third, and this seems to indeed be the case. Moreover, happiness doesn't seem to even show up as a yield icon at all, and no tile with 0 yield appears to be able to be worked. I have also tested the first two further and the +10 does in fact _not_ override the +1 from finding the natural wonders. So if you have the fountain of youth inside your borders, you have +11 happiness from it.

**Comment by yairm210** [comment]
That is all extremely counter intuitive. I think it makes for a worse gameplay experience, since it goes against everything the game has taught you so far. I'm against adding this to the game.

**Comment by SomeTroglodyte** [comment]
OK. I'd still like to know what `TileProvidesYieldWithoutPopulation` is actually used for.

**Pull request closed without merging**